### PR TITLE
[JENKINS-70199] Add distributed builds telemetry

### PR DESCRIPTION
**Pull request by daniel-beck**
https://issues.jenkins.io/browse/JENKINS-70199

This PR adds telemetry collecting information about whether/how distributed builds are set up at a fairly high level.

We generally consider distributed builds setups to be desirable, and encourage admins to set that up, but have no information about how common it is. While some of this is part of the regular submissions of usage stats, the way it's presented on stats.jenkins.io does not give us much useful information (e.g. total number does not tell us median and percentiles of numbers of agents). Additionally, this adds information about configured clouds, which isn't in the normal usage stats, as well as any dismissed related admin monitors. Number of jobs is collected to get an idea of instance size/complexity. Users would be nice but seems too expensive to get "actual" users (and not SCM committers) to be worth it.

### Testing done

Code seems straightforward enough and telemetry generally has test coverage, so I skipped specific autotests for this.

Via script console:

    ExtensionList.lookupSingleton(jenkins.telemetry.impl.DistributedBuilds.class).createContent()

Depending on the instance configuration, this resulted in the following output (`components` removed because that's a standard method and would clutter up sample output):

```
{"controllerExecutors":2,"controllerExecutorsWithAgentsWarning":true,"controllerExecutorsWithoutAgentsWarning":true,"clouds":{},"agents":{},"items":3,"components":{}}
```

```
{"controllerExecutors":0,"controllerExecutorsWithAgentsWarning":true,"controllerExecutorsWithoutAgentsWarning":true,"clouds":{"com.nirima.jenkins.plugins.docker.DockerCloud":1},"agents":{"hudson.slaves.DumbSlave":1},"items":3,"components":{}}
```

### Proposed changelog entries

- Add telemetry related to distributed builds

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7470"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
